src/Transport/HttpTransport.php

```diff
@@ -148,13 +146,19 @@ public function close(?int $timeout = null): PromiseInterface
      */
     private function cleanupPendingRequests(): void
     {
-        while ($promise = array_pop($this->pendingRequests)) {
-            try {
-                $promise->wait();
-            } catch (\Throwable $exception) {
-                // Do nothing because we don't want to break applications while
-                // trying to send events
-            }
+        try {
+            $requestGenerator = function () {
```

Comment:
Would you mind typehinting the return type please?

Reply:
Done